sklearn/utils/tests/test_validation.py

```diff
@@ -974,25 +976,54 @@ def test_check_is_fitted_with_attributes(wrap):
 
 
 def test_check_consistent_length():
+    """Test that `check_consistent_length` raises on inconsistent lengths and wrong
+    input types trigger TypeErrors."""
     check_consistent_length([1], [2], [3], [4], [5])
     check_consistent_length([[1, 2], [[1, 2]]], [1, 2], ["a", "b"])
     check_consistent_length([1], (2,), np.array([3]), sp.csr_matrix((1, 2)))
     with pytest.raises(ValueError, match="inconsistent numbers of samples"):
         check_consistent_length([1, 2], [1])
+
     with pytest.raises(TypeError, match=r"got <\w+ 'int'>"):
         check_consistent_length([1, 2], 1)
     with pytest.raises(TypeError, match=r"got <\w+ 'object'>"):
         check_consistent_length([1, 2], object())
-
     with pytest.raises(TypeError):
         check_consistent_length([1, 2], np.array(1))
-
     # Despite ensembles having __len__ they must raise TypeError
     with pytest.raises(TypeError, match="Expected sequence or array-like"):
         check_consistent_length([1, 2], RandomForestRegressor())
     # XXX: We should have a test with a string, but what is correct behaviour?
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, _", yield_namespace_device_dtype_combinations()
+)
+def test_check_consistent_length_array_API(array_namespace, device, _):
+    """Test that check_consistent_length works with different array types."""
+    xp = _array_api_for_tests(array_namespace, device)
+
+    check_consistent_length(
+        xp.asarray([1], device=device),
+        xp.asarray([2], device=device),
+    )
+    if xp.__name__ == "numpy":
+        check_consistent_length(
+            xp.asarray([[1, 2], [1, 2]], device=device),
+            xp.asarray([1, 2], device=device),
+            xp.asarray(["a", "b"], device=device),
+        )
+    else:
+        check_consistent_length(
+            xp.asarray([[1, 2], [1, 2]], device=device),
+            xp.asarray([1, 2], device=device),
+        )
+    with pytest.raises(ValueError, match="inconsistent numbers of samples"):
+        check_consistent_length(
+            xp.asarray([1, 2], device=device), xp.asarray([1], device=device)
+        )
```

Comment:
Since array API dispatch is disabled by default, all tests that check array API should enable the dispatch explicitly using the following context manager:

```python
with config_context(array_api_dispatch=True):
   ...
```

See other array API tests in other array API pull requests as example.